crates/ruff_python_formatter/src/comments/mod.rs

```diff
@@ -1,2 +1,274 @@
 #![allow(unused, unreachable_pub)] // TODO(micha): Remove after using the new comments infrastructure in the formatter.
+
+//! Types for extracting and representing comments of a syntax tree.
+//!
+//! Most programming languages support comments allowing programmers to document their programs.
+//! Comments are different from other syntax because programming languages allow comments in almost any position,
+//! giving programmers great flexibility on where they can write comments:
+//!
+//! ```javascript
+//! /**
+//!  * Documentation comment
+//!  */
+//! async /* comment */ function Test () // line comment
+//! {/*inline*/}
```

Comment:
In addition to placement, we also have "own-line" vs. "end-of-line" comments:

```python
def f():  # end-of-line
  # own-line
  pass
```

(Is that the correct terminology? Where and how is that behavior defined or handled?)

Reply:
Always thinking ahead. This is in the next PR

Reply:
Hah, yes, sorry, just came back here to comment that :)